GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -152,6 +152,35 @@ public void Execute()
             Environment.ExitCode = (int)this.ExitCode;
         }
 
+        protected virtual bool TryMountRepositories(out string consoleError)

```

Comment:
Should this be `abstract` in `UpgradeOrchestrator` and the implementation actually live in a `WindowsUpgradeOrchestrator`?

Reply:
Yes - ideally this should be the case. For the current state of this, this interacts with how the tests are architected. I am still looking into what it would take to move to this structure (in which case most of the logic would remain the same, but the implementation would slide down into a Windows specific type). If it introduces a lot of churn, I was going to defer this (and not the followup work item)